plugins/modules/vmware_vc_infraprofile_info.py

```diff
@@ -0,0 +1,292 @@
+
+- name: import Vcenter appliance infra profile config
+  vmware_vc_infraprofile_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: False
+    api: "export"
+    profiles: "ApplianceManagement"
+    config_path: "export.json"
+  delegate_to: localhost
+  '''
+
+RETURN = r'''
+    "list_infra":
+        description: A list of infra configs,
+        returned: on success with API as "list"
+        "sample": [
+            {
+                "info": "ApplianceManagement",
+                "name": "ApplianceManagement"
+            },
+            {
+                "info": "ApplianceNetwork",
+                "name": "ApplianceNetwork"
+            },
+            {
+                "info": "Authentication & Authorization Management",
+                "name": "AuthManagement"
+            }
+        ]
+
+    "export_infra":
+        description: A message about the exported file
+        returned: On success with API set as "export"
+        sample: {
+        "export_config_json":"json exported to file"
+        }
+
+    validate_infra:
+        description: A message about validate on exported file
+        returned: On success with API set as "validate"
+        "sample": {
+        "changed": false,
+        "failed": false,
+        "status": "VALID"
+        }
+
+    import_profile:
+        description: A message about import on import_profile spec
+        returned: On success with API set as "import"
+    "sample": {
+        "changed": true,
+        "failed": false,
+        "status": "0.0"
+    }
+'''
+
+import json
+import os
+import time
+
+
+class VcVersionChecker(PyVmomi):
+    def __init__(self, module):
+        super(VcVersionChecker, self).__init__(module)
+
+    def check_vc_version(self):
+        version = self.content.about.version
+        from distutils.version import LooseVersion
+        if LooseVersion(self.content.about.version) < LooseVersion('7'):
+            self.module.fail_json(msg="vCenter version is less than 7.0.0 Please specify vCenter with version greater than or equal to 7.0.0")
+
+
+class VcenterProfile(VmwareRestClient):
+
+    def __init__(self, module):
+        super(VcenterProfile, self).__init__(module)
+        self.config_path = self.params['config_path']
+
+    def list_vc_infraprofile_configs(self):
+        profile_configs_list = self.api_client.appliance.infraprofile.Configs.list()
+        config_list = []
+        for x in profile_configs_list:
+            config_list.append({'info': x.info, 'name': x.name})
+        self.module.exit_json(changed=False, infra_configs_list=config_list)
+
+    def get_profile_spec(self):
+        infra = self.api_client.appliance.infraprofile.Configs
+        profiles = {}
+        profiles = self.params['profiles'].split(",")
+        profile_spec = infra.ProfilesSpec(encryption_key="encryption_key", description="description", profiles=set(profiles))
```

Comment:
Why `encryption_key` and `description` don't have proper values?
I mean why they look like some test values and not anything meaningful

Reply:
its because, as we had a discussion with the API team they have not yet implemented this encryption and decryption mechanism in the back end, they only suggested using some test values not proper values as they don't have any implementation for this.